contracts/FluidityTap.sol

```diff
@@ -53,6 +53,17 @@ contract FluidityTap is ERC20Detailed, ERC20Burnable, ERC20Mintable, Ownable, Wh
         addWhitelisted(msg.sender); // ensure owner is whitelisted
     }
 
+    /**
+      * @notice Sets a new Treasury
+      * @dev Can only be performed by owner.
+      *
+      */
+    function updateTreasury(bytes9 _newCusip,  uint256 _newTotalFaceValue, uint256 _newMaturityDate) external onlyOwner {
+        cusip = _newCusip;
+        totalFaceValue = _newTotalFaceValue;
```

Comment:
Should this always be positive?

Reply:
not that i'm aware of. a yield can be negative but face value is still a positive value.